resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -43,6 +43,7 @@
    protected Map<String, Object> properties;
    protected ClientConfiguration configuration;
    protected byte[] bufferedEntity;
+   protected boolean streamRead;
    protected boolean streamFullyRead;

```

Comment:
While reviewing this PR, I noticed both the new streamRead and already existing streamFullyRead attributes are not volatile and perhaps should; comments? AFAICS, they're accessed from the InputStreamWrapper that's returned to users, so potentially we could have memory visibility issues, couldn't we?

Reply:
I think you're right.
User can invoke `Response.read(InputStream.class)` or just `Response.getEntity()` that return the `InputStreamWrapper` in the main thread  and decide to read it in another thread. Then after that when calling  `Response.getEntity()` or  `Response.bufferEntity()` in the main thread we can face a visibility issue. 
If every one OK i'll make the change.